### PR TITLE
Drop implicit alignment argument from FixedHash (and improve FunctionSelector helpers)

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/EVMHost.cpp [hunk 1]
```diff
@@ -266,7 +266,7 @@ evmc::result EVMHost::call(evmc_message const& _message) noexcept
 			bytes(begin(message.sender.bytes), end(message.sender.bytes)) +
 			bytes(begin(message.create2_salt.bytes), end(message.create2_salt.bytes)) +
 			keccak256(bytes(message.input_data, message.input_data + message.input_size)).asBytes()
-		));
+		), h160::AlignLeft);
```
Comment (axic) [inline test/EVMHost.cpp:269]: These in #13533 triggered the change. It is error prone having this implicitly, and the refactor revealed a lot of code duplication around `FunctionSelector`.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -1785,7 +1785,7 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 					""
 				);
 				define(IRVariable{_memberAccess}) << formatNumber(
-					util::selectorFromSignature(functionType.externalSignature())
+					util::selectorFromSignatureU256(functionType.externalSignature())
```
Comment (axic) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:1788]: We use a large variety of these helpers which could be further refactored/simplified. For example I doubt we need `formatNumber` here and instead should use `selectorFromSignatureH32().hex()`.

File: libsolidity/codegen/ArrayUtils.cpp [hunk 3]
```diff
@@ -889,7 +889,7 @@ void ArrayUtils::popStorageArrayElement(ArrayType const& _type) const
 			}
 			sstore(ref, slot_value)
 		})");
-		code("panicSelector", util::selectorFromSignature("Panic(uint256)").str());
+		code("panicSelector", util::selectorFromSignatureH32("Panic(uint256)").hex());
```
Comment (axic) [inline libsolidity/codegen/ArrayUtils.cpp]: This likely would be nicer with a `h32().hex()`. Should consider reviewing all instances in a separate PR.

File: test/EVMHost.cpp (hunk 1 above)
Comment (axic) [inline test/EVMHost.cpp:269]: These will be fixed in #13533.

File: test/libsolutil/FunctionSelector.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		util::selectorFromSignatureH32("test()"),
+		util::FixedHash<4>(0xf8a8fd6d)
+	);
+	BOOST_CHECK_EQUAL(
+		util::selectorFromSignatureU32("test()"),
+		4171824493
+	);
+	BOOST_CHECK_EQUAL(
+		util::selectorFromSignatureU256("test()"),
+		u256("112472165835992757616012344319229021616093528496533101562909820908405142847488")
```
Comment (cameel) [inline test/libsolutil/FunctionSelector.cpp]: Does `u256` accept hex strings?
```suggestion
		u256("0xf8a8fd6d00000000000000000000000000000000000000000000000000000000")
```

File: test/libsolutil/FunctionSelector.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+BOOST_AUTO_TEST_CASE(conversions)
+{
+	BOOST_CHECK_EQUAL(
+		util::selectorFromSignatureH32("test()"),
+		util::FixedHash<4>(0xf8a8fd6d)
+	);
+	BOOST_CHECK_EQUAL(
+		util::selectorFromSignatureU32("test()"),
+		4171824493
```
Comment (cameel) [inline test/libsolutil/FunctionSelector.cpp]:
```suggestion
		0xf8a8fd6d
```

File: test/EVMHost.cpp (hunk 1 above)
Comment (cameel) [reply test/EVMHost.cpp:269]: What will be fixed? Doesn't this just preserve the original behavior by using the default here?

**Review by cameel — DISMISSED** [review]

**Review by leonardoalt — DISMISSED** [review]

File: test/EVMHost.cpp (hunk 1 above)
Comment (axic) [reply test/EVMHost.cpp:269]: This PR preserves current behaviour. The code will be "fixed" to be more accurate.

File: test/libsolutil/FunctionSelector.cpp (hunk 4 above)
Comment (axic) [reply test/libsolutil/FunctionSelector.cpp]: Changed.

File: test/EVMHost.cpp (hunk 1 above)
Comment (cameel) [reply test/EVMHost.cpp:269]: Ah, ok.

**Review by cameel — APPROVED** [review]

**Pull request merged**